frontend/src/lib/components/AnnotationsOverlay/annotationsOverlayLogic.ts

```diff
@@ -129,18 +141,21 @@ export const annotationsOverlayLogic = kea<annotationsOverlayLogicType>([
             },
         ],
         relevantAnnotations: [
-            (s, p) => [s.annotations, s.dateRange, p.insightNumericId],
-            (annotations, dateRange, insightNumericId) => {
+            (s, p) => [s.annotations, s.dateRange, p.insightNumericId, p.dashboardId],
+            (annotations, dateRange, insightNumericId, dashboardNumericId) => {
                 // This assumes that there are no more annotations in the project than AnnotationsViewSet
                 // pagination class's default_limit of 100. As of June 2023, this is not true on Cloud US,
-                // where 3 projects exceed this limit. To accomodate those, we should always make a request for the
+                // where 3 projects exceed this limit. To accommodate those, we should always make a request for the
                 // date range of the graph, and not rely on the annotations in the store.
                 return (
                     dateRange
                         ? annotations.filter(
                               (annotation) =>
                                   (annotation.scope !== AnnotationScope.Insight ||
                                       annotation.dashboard_item === insightNumericId) &&
+                                  (annotation.scope !== AnnotationScope.Dashboard ||
+                                      annotation.dashboard === dashboardNumericId ||
+                                      annotation.dashboard_item === insightNumericId) &&
```

Comment:
This makes for an element of surprise when clicking into an insight from a dashboard: dashboard-scoped annotations disappear. I thought we were going the other way – i.e. the insight page showing annotations from all dashboard this insight is on

https://github.com/user-attachments/assets/1031f5c6-a9d1-4d07-a160-2f77d290a7e7

Reply:
Guess I misunderstood earlier, as on clicking in you might see more annotations than were originally on the insight (if it's on other dashboards with dashboard scoped insights).
But will change as you suggest as the above video is the more common scenario.